src/components/OrderLimitPopover.vue

```diff
@@ -0,0 +1,114 @@
+      </ion-item>
+      <ion-item button @click="updateOrderLimitType('no-capacity')">
+        {{ $t("No Capacity") }}
+        <ion-icon slot="end" :icon="lockClosedOutline" />
+      </ion-item>
+      <ion-item button lines="none" @click="updateOrderLimitType('custom')">
+        {{ $t("Custom") }}
+      </ion-item>
+    </ion-list>
+  </ion-content>
+</template>
+  
+<script lang="ts">
+import {
+  IonContent,
+  IonIcon,
+  IonItem,
+  IonList,
+  IonListHeader,
+  alertController,
+  popoverController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { lockClosedOutline, lockOpenOutline } from 'ionicons/icons'
+import { translate } from '@/i18n';
+import { showToast } from '@/utils';
+export default defineComponent({
+  name: "OrderLimitPopover",
+  components: {
+    IonContent,
+    IonIcon,
+    IonItem,
+    IonList,
+    IonListHeader
+  },
+  setup() {
+    return {
+      lockClosedOutline,
+      lockOpenOutline
+    }
+  },
+  props: ['fulfillmentOrderLimit'],
+  data() {
+    return {
+      setLimit: this.fulfillmentOrderLimit as any
+    }
+  },
+  methods: {
+    async updateOrderLimitType(orderLimitType: string) {
+      let header = "Unlimited fulfillment capacity"
+      let message = "Unlimited capacity removes the fulfillment capacity limit entirely. To add a fulfillment capacity to this facility, use the custom option."
+      let showInput = false
+
+      if (orderLimitType === 'custom') {
+        header = "Custom fulfillment capacity"
+        message = ""
+        showInput = true
+      } else if (orderLimitType === 'no-capacity') {
+        this.setLimit = 0
+        header = "No fulfillment capacity"
+        message = "No capacity removes sets the fulfillment capacity to 0, preventing any new orders from being allocated to this facility. Use the \"Reject all orders\" option in the fulfillment pages to clear your facilities fulfillment queue. To add a fulfillment capacity to this facility, use the custom option."
+      } else if (orderLimitType === 'unlimited') {
+        this.setLimit = ""
+      }
+
+      this.showOrderLimitAlert(header, message, showInput)
+    },
+    async showOrderLimitAlert(header: string, message: string, showInput: boolean) {
+      const alert = await alertController.create({
+        header: translate(header),
+        message: translate(message, {space: '</br></br>'}),
+        inputs: showInput ? [{
+          name: "setLimit",
+          placeholder: translate("Order fulfillment capacity"),
+          type: "number",
+          min: 0
+        }] : [],
+        buttons: showInput ? [{
+          text: translate('Cancel'),
+          role: "cancel"
+        },
+        {
+          text: translate('Apply'),
+          handler: (data) => {
+            if (data.setLimit <= 0) {
+              showToast(translate('Provide a value greater than 0'))
+              return;
+            }
+            popoverController.dismiss(data.setLimit)
```

Comment:
We can improve this buttons to define them once and add the logic on apply button handler based on whether the input is displayed or not.